lib/irb/cmd/history.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "stringio"
+require_relative "nop"
+require_relative "../pager"
+
+module IRB
+  # :stopdoc:
+
+  module ExtendCommand
+    class History < Nop
+      category "IRB"
+      description "Show the input history."
+
+      def execute(*)
+        output = StringIO.new
+        irb_context.io.class::HISTORY.each_with_index.reverse_each do |input, index|
+          header = "#{index}: "
+          header_length = header.size
+
+          lines = input.split("\n")
+          first_line = header + lines[0]
```

Comment:
If input is an empty string, `lines[0]` will be `nil`.

In some rare case, HISTORY could contain empty string.
```ruby
$ irb --nomultiline
irb(main):001> Readline.readline '>', true
> (press enter)
=> ""
irb(main):002> irb_context.io.class::HISTORY.to_a.last(2)
=> ["", "irb_context.io.class::HISTORY.to_a.last(2)"]
```
 When this history is once saved to `.irb_history`, HISTORY will always contain empty string. (I have "" in my .irb_history)

Reply:
Hopefully https://github.com/ruby/irb/pull/761/commits/214fc8acd46976e10aa74a36068342354343e95d handles that issue